client/src/components/ReportWorkflow.js

```diff
@@ -187,3 +188,39 @@ export const ReportCompactWorkflow = ({ workflow }) => (
 ReportCompactWorkflow.propTypes = {
   workflow: PropTypes.array.isRequired
 }
+
+export const CompactRowReportWorkflow = ({ workflow, compactStyle }) => (
+  <Fieldset
+    className="workflow-fieldset compact"
+    title="Workflow"
+    compactStyle={compactStyle}
+  >
+    <CompactRow
+      rowType={ROW_TYPES.onlyData}
+      style={compactStyle}
+      content={workflow.map(action => {
+        const key = action.step
+          ? `${action.createdAt}-${action.step.uuid}`
+          : action.createdAt
+        return <CompactRowReportAction action={action} key={key} />
+      })}
+    />
+  </Fieldset>
+)
+
+CompactRowReportWorkflow.propTypes = {
+  workflow: PropTypes.array.isRequired,
+  compactStyle: PropTypes.string
+}
+
+const CompactRowReportAction = ({ action }) => {
```

Comment:
Instead of defining a new component it would be better to use existing `ReportAction` component by adding a prop `hideDetails` which defaults to `false`.

Reply:
I tried this method first, the component becomes much more complicated with conditional returns etc. I need to change 3 components and each needs to return compact version conditionally etc. I thin having an extra component here will be fine.